pytorch_lightning/trainer/training_loop.py

```diff
@@ -140,7 +140,7 @@ def on_train_end(self):
         # todo: TPU 8 cores hangs in flush with TensorBoard. Might do for all loggers.
         # It might be related to xla tensors blocked when moving the cpu
         # kill loggers
-        if self.trainer.logger is not None and self.trainer.training_type_plugin.should_finalize:
```

Comment:
aha!

Reply:
Yes, closing the logger solves the issue.